backend/openapi.yaml

```diff
@@ -4881,22 +5097,46 @@ components:
           example: 42
         credentials_label:
           type: string
-          example: "ogp-twilio-creds"
+          example: 'ogp-twilio-creds'
         recipient:
           type: string
-          example: "98765432"
+          example: '98765432'
         body:
           type: string
-          example: "Hello world"
-        message_id:
-          type: string
-          example: "SM12345678901234567890123456789012"
+          example: 'Hello world'
         created_at:
           type: string
           format: date-time
         updated_at:
           type: string
           format: date-time
+        accepted_at:
+          type: string
+          format: date-time
+        sent_at:
+          type: string
+          format: date-time
+        delivered_at:
+          type: string
+          format: date-time
+        errored_at:
+          type: string
+          format: date-time
+        error_code:
+          type: string
+        status:
+          type: string
+          enum: [UNSENT, ACCEPTED, SENT, DELIVERED, ERROR]
+          description: >
```

Comment:
**[question/suggestion]** how are we aligning these with [the statuses supported by Twilio](https://support.twilio.com/hc/en-us/articles/223134347-What-are-the-Possible-SMS-and-MMS-Message-Statuses-and-What-do-They-Mean-)? maybe we can link to the article in the description here? some notes:
- some SMSes that are actually delivered may be stuck at `SENT` because the telcos don't provide callbacks
- I think we can just put `failed` and `undelivered` under `ERROR` and put more info in the `error_code` field? 

**[thinking out loud]** not sure whether these are the same as the ones in the camapgin delivery reports, but don't think aligning with that is that important

**[change request]** there are some typos below using email instead of SMS haha

**[nit]** would be good to use `SMS` instead of `sms`

Reply:
ERROR is for both `failed` and `undelivered` in the current design. What I'm doing is exactly what u're suggesting above

Reply:
think it's good to add a link to the Twilio, so the implementation/API user will know